### PR TITLE
[WIP] catalog in meshery ui

### PR DESCRIPTION
**Pull request by warunicorn19**
Signed-off-by: Aditya Chatterjee <speak2adi@gmail.com>

**Description**
This is just to get the catalog UI started. 
Screenshot 2022-08-13 at 3 59 55 PM

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by gr455** [comment]
@warunicorn19 Can we have logic in the UI to mark patterns that have visibility=public and userID != currentUser.ID as catalog items?

**Comment by warunicorn19** [comment]
I dont think that should be a problem 🤔. @Chadha93 @MUzairS15 mKes sense right?

**Comment by gr455** [comment]
> I dont think that should be a problem thinking. @Chadha93 @MUzairS15 makes sense right?

There is still a little proble with pagination. If we want to show all the catalog items together, we can't just query them with the other items given that they are paginated :thinking:

File: ui/components/MesheryPatterns.js [hunk 1]
```diff
@@ -626,6 +626,29 @@ function MesheryPatterns({
...
+          return (
+            <TableCell key={index}>
+              <b>{column.label}</b>
+            </TableCell>
+          );
+        },
+        customBodyRender : function CustomBody() {
+          return (
+            <>
+              <img src="/static/img/private.svg" />
```
Comment (Chadha93) [inline ui/components/MesheryPatterns.js:646]: This should be dynamically referenced. Assuming all we have a logic defined to select the visibility. On that basis, we can show private or public. cc @gr455

**Comment by Chadha93** [comment]
> > I dont think that should be a problem thinking. @Chadha93 @MUzairS15 makes sense right?
> 
> There is still a little proble with pagination. If we want to show all the catalog items together, we can't just query them with the other items given that they are paginated thinking

@gr455 Is there a variable or logic defined in the backend or cloud, to get the visibility type? Like we did for source-type.

**Comment by gr455** [comment]
> > > I dont think that should be a problem thinking. @Chadha93 @MUzairS15 makes sense right?
> > 
> > There is still a little proble with pagination. If we want to show all the catalog items together, we can't just query them with the other items given that they are paginated thinking
> 
> @gr455 Is there a variable or logic defined in the backend or cloud, to get the visibility type? Like we did for source-type.

I think I'll need to add it to graphql still.

**Comment by warunicorn19** [comment]
> > > I dont think that should be a problem thinking. @Chadha93 @MUzairS15 makes sense right?
> 
> > 
> 
> > There is still a little proble with pagination. If we want to show all the catalog items together, we can't just query them with the other items given that they are paginated thinking
> 
> 
> 
> @gr455 Is there a variable or logic defined in the backend or cloud, to get the visibility type? Like we did for source-type.

Yup I know I've just added that as a placeholder.

**Comment by gr455** [comment]
@MUzairS15 Merge conflicts :disappointed: 
Should we merge https://github.com/meshery/meshery/pull/6028 ?

**Comment by MUzairS15** [comment]
@warunicorn19 @warunicorn19 Can you please test and report any left items??

**Comment by MUzairS15** [comment]
> @MUzairS15 Merge conflicts 😞
> Should we merge #6028 ?

Yes, will resolve them

**Comment by leecalcote** [comment]
> @warunicorn19 @warunicorn19 Can you please test and report any left items??

Aditya is away from keyboard this weekend. Will need to move forward in his absence.

**Comment by gr455** [comment]
Currently the UI does not show the public items. FYI the catalog items are not returned by `/api/patterns`. You will have to make a request to `/api/patterns/catalog`.

**Comment by MUzairS15** [comment]
> Currently the UI does not show the public items. FYI the catalog items are not returned by `/api/patterns`. You will have to make a request to `/api/patterns/catalog`.

Actually, I tested yesterday they were shown. I had merged server PR locally and when I dropped that commit to make this UI commit, I don't know why again there were conflicts, and those changes I think didn't appear,

**Comment by gr455** [comment]
> > Currently the UI does not show the public items. FYI the catalog items are not returned by `/api/patterns`. You will have to make a request to `/api/patterns/catalog`.
> 
> Actually, I tested yesterday they were shown. I had merged server PR locally and when I dropped that commit to make this UI commit, I don't know why again there were conflicts, and those changes I think didn't appear,

Oh, I resolved these current conflicts locally and merged into the server PR and tried, and the public items didn't appear. Will you re-add those changes?

**Comment by MUzairS15** [comment]
@Chadha93 @gr455 Check now

**Comment by gr455** [comment]
@MUzairS15 Still now showing public items.
Also one more thing I wanted to point out is that `/api/pattern` may also return public patterns if the pattern is owned by the user (although for now that will never be the case, but it's better to account for that now itself).

Right now I don't see public items owned by me or by any other user. Lee has added some catalog items to the prod database also, so you can test it with that also.

**Comment by gr455** [comment]
> @MUzairS15 Still now showing public items. Also one more thing I wanted to point out is that `/api/pattern` may also return public patterns if the pattern is owned by the user (although for now that will never be the case, but it's better to account for that now itself).
> 
> Right now I don't see public items owned by me or by any other user. Lee has added some catalog items to the prod database also, so you can test it with that also.

Wait, I see it on production database. I might have been doing something wrong on local

**Comment by Chadha93** [comment]
@gr455 Are the seeded patterns/filters will be considered private?

**Comment by leecalcote** [comment]
@Chadha93, no, the seeded Local Provider content is considered public.

**Comment by gr455** [comment]
> @gr455 Are the seeded patterns/filters will be considered private?

Yes

**Comment by gr455** [comment]
Uh oh. I thought we discussed those to be private

**Comment by Chadha93** [comment]
> 

I think none is available for all, and the content there should always be public.

**Comment by gr455** [comment]
> > 
> 
> I think none is available for all, and the content there should always be public.

But each instance of Meshery can have only one None user right. So public or private, it's only accessible by one user. So sure, we can make it public; not a big deal and functionally no change

**Comment by Chadha93** [comment]
> > > 
> > 
> > 
> > I think none is available for all, and the content there should always be public.
> 
> But each instance of Meshery can have only one None user right. So public or private, it's only accessible by one user. So sure, we can make it public; not a big deal and functionally no change

Yepp :ok_hand:

**Comment by warunicorn19** [comment]
Screenshot 2022-08-31 at 12 38 22 AM

i feel like this looks better.

**Comment by warunicorn19** [comment]
Let's merge

**Pull request merged**